win32ss/gdi/gdi32/objects/text.c

```diff
@@ -19,17 +26,25 @@ TextOutA(
     UNICODE_STRING StringU;
     BOOL bResult;
 
-    if (lpString != NULL)
+    if (lpString != NULL && cchString > 0)
     {
-        RtlInitAnsiString(&StringA, (LPSTR)lpString);
+        if (cchString > MAXUSHORT - sizeof(CHAR))
+            cchString = MAXUSHORT - sizeof(CHAR);
+
+        StringA.Length = (USHORT)cchString;
+        StringA.MaximumLength = (USHORT)cchString + 1;
+        StringA.Buffer = (PCHAR)lpString;
+
         RtlAnsiStringToUnicodeString(&StringU, &StringA, TRUE);
```

Comment:
This could use a check for failure while you're here. But the change you have looks good so you can also leave it as is.

Reply:
OK. Check was added.